daras_ai_v2/language_model.py

```diff
@@ -743,7 +771,7 @@ def _run_chat_model(
             return _run_openai_chat(
                 model=model,
                 avoid_repetition=avoid_repetition,
-                max_tokens=max_tokens,
+                max_completion_tokens=max_tokens,
```

Comment:
https://platform.openai.com/docs/guides/reasoning#controlling-costs

Reply:
it has been deprecated completely (errors with max_tokens). + max_tokens = max_completion_tokens for models that are not o1

```
To manage costs with the o1 series models, you can limit the total number of tokens the model generates (including both reasoning and completion tokens) by using the [max_completion_tokens](https://platform.openai.com/docs/api-reference/chat/create#chat-create-max_completion_tokens) parameter.

In previous models, the max_tokens parameter controlled both the number of tokens generated and the number of tokens visible to the user, which were always equal. However, with the o1 series, the total tokens generated can exceed the number of visible tokens due to the internal reasoning tokens.

Because some applications might rely on max_tokens matching the number of tokens received from the API, the o1 series introduces max_completion_tokens to explicitly control the total number of tokens generated by the model, including both reasoning and visible completion tokens. This explicit opt-in ensures no existing applications break when using the new models. The max_tokens parameter continues to function as before for all previous models.
```

Reply:
https://platform.openai.com/docs/api-reference/chat/create#chat-create-max_tokens